pkg/kapis/devops/v1alpha2/register.go

```diff
@@ -226,6 +226,16 @@ func AddPipelineToWebService(webservice *restful.WebService, devopsClient devops
 			Returns(http.StatusOK, "The filed of \"Url\" in response can download artifacts", []devops.Artifacts{}).
 			Writes([]devops.Artifacts{}))
 
+		// download PipelineRun artifact
+		webservice.Route(webservice.GET("/devops/{devops}/pipelines/{pipeline}/runs/{run}/artifacts/download").
```

Comment:
I prefer to use `pipelinerun` instead of `run`. We don't have a CRD named `run`.

Reply:
Or, `/devops/{devops}/pipelineruns/{pipelinerun}/artifacts/download` is good too.

Reply:
Given the CRD `PipelineRun` version is `v1alpha3`. So, I agree with the version change.

But I'm ok with both the following APIs:
* `/namespaces/{namespace}/pipelineruns/{pipelinerun}/artifacts/download`
* `/devops/{devops}/pipelines/{pipeline}/pipelineruns/{pipelinerun}/artifacts/download`

hi @JohnNiang , what do you think about it?

Reply:
I'd prefer to `/devops/{devops}/pipelines/{pipeline}/pipelineruns/{pipelinerun}/artifacts/download`.

Reply:
I looked at the existing code, `v1alpha2` has not such thing as `pipelineruns/{pipelinerun}`. 

then using `/devops/{devops}/pipelines/{pipeline}/pipelineruns/{pipelinerun}/artifacts/download` suggest version change.
And this`pipelinerun` is different from the`pipelineruns` of `v1alpha3`

Reply:
> 

`pipelineruns` of `v1alpha3` is CRD `pipelineName`.
So. for v1alpha3. i prefer to use `/namespaces/{namespace}/pipelineruns/{pipelinerun}/artifacts/download`

Reply:
I found we have configured `role-template-view-pipelineruns` role in [here](https://github.com/kubesphere/ks-installer/blob/1c8d2181f2fe1cb24aaefce96ae0e390edf9597e/roles/ks-core/prepare/files/ks-init/role-templates.yaml#L3279-L3308) which contains `pipelineruns` and `pipeilnes/pipelienruns`.

So I'm ok to `/namespaces/{namespace}/pipelineruns/{pipelinerun}/artifacts/download` as well.

Reply:
Any updates here?

Reply:
> Any updates here?

Changes will be committed later